virttest/qemu_vm.py

```diff
@@ -3577,8 +3577,21 @@ def activate_nic(self, nic_index_or_name):
             else:
                 raise virt_vm.VMAddNicError("No free root port for device %s"
                                             " to plug." % nic.nic_name)
-        error_context.context("Activating nic on VM %s with monitor command %s" % (
-            self.name, device_add_cmd))
+        elif isinstance(bus, qdevices.QPCIBus):
+            nic_dev = qdevices.QDevice(params={'id': nic.nic_name,
+                                               'driver': nic.nic_model},
+                                       parent_bus=({'aobject': 'pci.0'},))
+            nic_dev.set_aid(nic_dev.get_qid())
+            qdev_out = bus.insert(nic_dev)
+            if not isinstance(qdev_out, str):
+                device_add_cmd += ",bus=%s,addr=%s" % (nic_dev.get_param('bus'),
+                                                       nic_dev.get_param('addr'))
+            else:
+                raise virt_vm.VMAddNicError("%s for device %s to plug."
+                                            % (qdev_out, nic.nic_name))
+
+        error_context.context("Activating nic on VM %s with monitor command %s"
+                              % (self.name, device_add_cmd))
```

Comment:
Seems we can remove `QPCIEBus` part, it can be unify to your code, since`QPCIEBus.insert()` will perform similar function for L#3571-3576

Reply:
OK, got it. And ppc does not use PCIE, I need a little time to try and update.

Reply:
@vivianQizhu Updated according to your suggestion, could you review again?